File: 462.最少移动次数使数组元素相等-ii.go
/*
 * @lc app=leetcode.cn id=462 lang=golang
 *
 * [462] 最少移动次数使数组元素相等 II
 */

// @lc code=start
func minMoves2(nums []int) (res int) {
	sort.Ints(nums)
	middleNum := nums[len(nums)/2]
	for _, num := range nums {
		res += abs(num - middleNum)
	}
	return
}

func abs(num int) int {
	if num >= 0 {
		return num
	}
	return -num
}

// @lc code=end

File: 462.最少移动次数使数组元素相等-ii.py
#
# @lc app=leetcode.cn id=462 lang=python3
#
# [462] 最少移动次数使数组元素相等 II
#

# @lc code=start
class Solution:
    def minMoves2(self, nums: List[int]) -> int:
        nums.sort()
        middle_num = nums[len(nums) // 2]
        return sum(abs(num - middle_num) for num in nums)
# @lc code=end

